### PR TITLE
feat: migrate to buf managed mode

### PR DESCRIPTION
**Pull request by odsod**

### CODE REVIEW

File: proto/Makefile [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+include tools/clang-format/rules.mk
+include tools/protoc-gen-go-grpc/rules.mk
+
+build/protoc-gen-go: ../go.mod
+	$(info [$@] building binary...)
+	@go build -o $@ google.golang.org/protobuf/cmd/protoc-gen-go
+
+build/protoc-gen-go-aip: ../go.mod
+	$(info [$@] building binary...)
+	@go build -o $@ go.einride.tech/aip/cmd/protoc-gen-go-aip
```
Comment (viktorvoltaire) [inline proto/Makefile:23]: So this is to build what we would normally code generate in the proto repo?

File: proto/Makefile [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+.PHONY: buf-lint
+buf-lint: $(buf)
+	$(info [$@] linting proto schemas...)
+	@$(buf) lint
+
+.PHONY: buf-breaking
+buf-breaking: $(buf)
+	$(info [$@] checking for breaking changes against origin/master)
+	@$(buf) breaking --against buf.build/einride/iam --path einride
```
Comment (viktorvoltaire) [inline proto/Makefile:43]: :tada:

File: proto/einride/iam/v1/annotations.proto [hunk 3]
```diff
@@ -2,14 +2,6 @@ syntax = "proto3";
 
 package einride.iam.v1;
 
-option csharp_namespace = "Iam.V1";
```
Comment (viktorvoltaire) [inline proto/einride/iam/v1/annotations.proto:5]: Are these all removed by the linter? Same goes for the other proto files

**Review by viktorvoltaire — APPROVED** [review]
LGTM

File: proto/Makefile (hunk 1 above)
Comment (odsod) [reply proto/Makefile:23]: this is to build the `protoc-gen-go-aip` code generator plugin - we build it like this to ensure we build it with the same version as the repo depends on

File: proto/einride/iam/v1/annotations.proto (hunk 3 above)
Comment (odsod) [reply proto/einride/iam/v1/annotations.proto:5]: They are no longer needed when using managed mode. With managed mode, the file options are configured as part of the code generation config and not the schema definition.

See here:

https://docs.buf.build/tour/use-managed-mode/

Comment (viktorvoltaire) [reply proto/einride/iam/v1/annotations.proto:5]: Sweet!

**Pull request merged**